R/utils.R

```diff
@@ -14,3 +14,28 @@ list_repo <- function(orgname = 'natverse') {
   repos <- gh::gh(paste0("/orgs/",orgname,"/repos"), type = "public")
   vapply(repos, "[[", "", "name")
 }
+
+have_xquartz <- function() {
+  isTRUE(nzchar(Sys.which('Xquartz')))
+}
+
+system_requirements_ok <- function() {
+  if(grepl("linux-gnu", R.version$os) | grepl("mingw", R.version$os)) {
+    return(TRUE)
+  }
+  if(grepl("^darwin", R.version$os)) {
+    if(!have_xquartz()){
+      usethis::ui_todo(paste("Please download and install Xquartz from",
+                       "https://www.xquartz.org!\nThis is a system requirement",
+                       "and is needed for 3D display of neurons."))
+      return(FALSE)
+    }
+    return(TRUE)
+  }
+  if(isTRUE(.Platform$OS.type == "windows")) {
+    return(TRUE)
+  }
```

Comment:
I think this line is not required as `grepl("mingw", R.version$os)` will return TRUE for windows operating system. One of the reasons I wanted to use this logic is, R.version$os can be used for detecting
1. Windows OS - `grepl("mingw", R.version$os)`
2. Mac OS - `grepl("^darwin", R.version$os)`
3. Linux OS - `grepl("linux-gnu", R.version$os)`

Hence we can avoid using `.Platform$OS.type`

Reply:
Oh I see. Thanks. Thing is I have never seen any function check that to determine if R is running on Windows and I think there may well be a reason why that's the case. Checking .Platform$OS.type doesn't come at any cost and I think it would be better to have separate code blocks for each OS so that we can plug in the actual requirements.

Reply:
Talking of which, were there any requirements on linux for rgl to work?

Reply:
I agree with your idea to have separate code blocks for each OS, coming back to your question of using `R.version$os` to identify windows you can see examples where people have posted their `R.version` for windows while asking other questions (hence i think `mingw` can be reliably used):
1) https://stackoverflow.com/questions/59251390/how-to-install-x11-in-r-on-windows-10
2) https://stackoverflow.com/questions/61093953/how-to-view-change-the-installation-of-r-that-rstudio-uses

For the `rgl` in linux I haven't seen any specific requirement (it used to work a while back when I had linux workstation), but may be someone else in the lab using linux can pitch in?

Reply:
A more reliable answer for `mingw` can be found in the source code of `R` itself:
https://github.com/wch/r-source/blob/9418fc2c57b181d7a38f26520f8bd75ddc8488be/src/gnuwin32/fixed/h/config.h#L1090

Reply:
(left over comment, I think you did a similar kind of search the other way!)
Yes but `.Platform$OS.type == "windows"` is used in many places in the core R source code and I can only find one test for mingw which is for a build architecture check.

https://github.com/wch/r-source/search?q=.Platform%24OS.type+%3D%3D+%22windows%22&unscoped_q=.Platform%24OS.type+%3D%3D+%22windows%22